src/main/java/com/CarRentalAgency/services/DealerServiceImpl.java

```diff
@@ -42,13 +42,11 @@ public Dealer saveDealer(Dealer dealer) {
         // check if the dealer with the same email already exists
         Dealer existingDealer = dealerRepository
                 .findByEmail(dealer.getEmail()) // return the dealer with the specified email
-                .orElse(null); // if the dealer does not exist return null
+                .orElseThrow(() -> new AlreadyExistsException("THIS EMAIL:" + dealer.getEmail() + " ALREADY USED !!"));
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new implementation throws an exception if a dealer with the same email already exists. However, it doesn't assign the result to `existingDealer` as in the old code. This might lead to unexpected behavior if `existingDealer` is used later in the code. To maintain the original logic while improving error handling, you should still assign the result to `existingDealer`.

```diff
- .orElseThrow(() -> new AlreadyExistsException("THIS EMAIL:" + dealer.getEmail() + " ALREADY USED !!"));
+ existingDealer = .orElseThrow(() -> new AlreadyExistsException("THIS EMAIL:" + dealer.getEmail() + " ALREADY USED !!"));
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->